pymc/model.py

```diff
@@ -2035,10 +2035,74 @@ def Deterministic(name, var, model=None, dims=None):
+      A soft constraint is a constraint that is only partially satisfied. The effect of this is that the posterior probability for the parameters decreases as they move away from the constraint, but does not become exactly zero.
+      This allows the sampler to generate values that violate the constraint, but with lower probability.
+
+    .. code:: python
+
+        with pm.Model() as model:
+          x = pm.Normal('x', mu=0.1, sigma=1)
+          y = pm.Normal('y', mu=x, sigma=1, observed=data)
+          constraint = x>=0
+          potential = pm.Potential('x_constraint', pm.math.log(pm.math.switch(constraint, 1, 0.5)))
+
+      In this example, Potential is used to obtain an arbitrary prior. This prior distribution refers to the prior knowledge that the values of max_items are likely to be small rather than being large.
+      The prior probability of max_items is defined using a Potential object with the log of the inverse of max_items as its value.
+      This means that larger values of max_items have a lower prior probability density, while smaller values of max_items have a higher prior probability density.
+      When the model is sampled, the posterior distribution of ``max_items`` given the observed value of ``n_items`` will be influenced by the power-law prior defined in the Potential object
+
+     .. code:: python
+
+        with pm.Model():
+          # p(max_items) = 1 / max_items
+          max_items = pm.Uniform("max_items", lower=1, upper=100)
+          pm.Potential('power_prior', pm.math.log(1/max_items))
+
+          n_items = pm.Uniform('n_items', lower=1, upper=max_items, observed=60)
+
+      In the next example, the soft_sum_constraint potential encourages x and y to have a small sum, effectively adding a soft constraint on the relationship between the two variables.
+      This can be useful in cases where you want to ensure that the sum of multiple variables stays within a certain range, without enforcing an exact value.
+      In this case, the larger the deviation, larger will be the negative value (-((x + y)**2)) which the MCMC sampler will attempt to minimize. However, the sampler might generate values for some small deviations but with lower probability hence this is a soft constraint.
+
+     .. code:: python
+
+        with pm.Model() as model:
+          x = pm.Normal('x', mu=0.1, sigma=1)
+          y = pm.Normal('y', mu=x, sigma=1, observed=data)
+          soft_sum_constraint = pm.Potential("soft_sum_constraint", -((x + y)**2))
+
+      The potential value is incorporated into the model log-probability, so it should be -inf (or very negative) when a constraint is violated, so that those draws are rejected. 0 won't have any effect and positive values will make the proposals more likely to be accepted
```

Comment:
All this should go at the bottom of the docstring and have a section header too. Ref: https://numpydoc.readthedocs.io/en/latest/format.html#examples